lasagne/layers/pool.py

```diff
@@ -15,81 +14,212 @@
 ]
 
 
-def pool_output_length(input_length, pool_length, ignore_border=True):
-    '''Compute the output length of a pooling operator
-    along a particular dimension.
+def pool_output_length(input_length, ds, st, ignore_border=True, pad=0):
+    """
+    Compute the output length of a pooling operator
+    along a single dimension.
 
     Parameters
     ----------
-    input_length
-    pool_length
-        Shape of the input and pooling operator in the chosen dimension
-
-    ignore_border:
-        if True, the output length is rounded down.
-        if False, it is rounded up
+    input_length : integer
+        The length of the input in the pooling dimension
+    ds : integer
+        The length of the pooling region
+    st : integer
+        The stride between successive pooling regions
+    pad : integer
+        The number of elements to be added to the input on each side.
+    ignore_border: bool
+        If True, partial pooling regions will be ignored.
 
     Returns
     -------
     output_length
-        * None if either input is None
-        * `input_length / pool_length` otherwise
-    '''
-
-    if input_length is None or pool_length is None:
+        * None if either input is None.
+        * Computed length of the pooling operator otherwise.
+
+    Notes
+    -----
+    When ignore_border = True, this is given by the number of full
+    pooling regions that fit in the padded input length,
+    divided by the stride (rounding down).
+
+    When ignore_border = False, partial pooling regions are also
+    allowed, as long as each pooling region contains at least one element
+    not present in another pooling region.
```

Comment:
I'm a bit confused about that definition. I think it's because it speaks of partial pooling regions (in plural). What about:
```
For `ignore_border=False`, a single partial pooling region is appended if at least one input element would be left uncovered otherwise.
```